packages/components/time-select/src/time-select.vue

```diff
@@ -48,8 +48,13 @@ interface Time {
 const parseTime = (time: string): null | Time => {
   const values = (time || '').split(':')
   if (values.length >= 2) {
-    const hours = parseInt(values[0], 10)
+    let hours = parseInt(values[0], 10)
     const minutes = parseInt(values[1], 10)
+    if (time.includes('AM') && hours === 12) {
+      hours = 0
+    } else if (time.includes('PM') && hours !== 12) {
+      hours += 12
```

Comment:
**Suggestion:**  Can we do a safer check here?

e.g.

```ts
time = time.toLowerCase() // or toUpperCase()

if (time.includes('am') && hours = 12) { // or includes('AM')
}
```

Reply:
ok